kube-runtime/src/reflector/object_ref.rs

```diff
@@ -28,10 +31,11 @@ use std::{
 ///     ObjectRef::<Secret>::new("a").erase(),
 /// );
 /// ```
+#[non_exhaustive]
 pub struct ObjectRef<K: Resource> {
-    dyntype: K::DynamicType,
+    pub dyntype: K::DynamicType,
     /// The name of the object
-    pub name: String,
+    pub name: Option<String>,
```

Comment:
ah, you option wrapped name?

Reply:
I still think it's the way to go long-term, but this is the wrong PR to do it in.